### ISSUE TITLE
az aks upgrade --kubernetes-version leaves cluster in bad state

### ISSUE THREAD

**Issue by wasker** [Azure/azure-cli#22928]
**Related command**

az aks upgrade --kubernetes-version --yes -g myrg -n myaks

**Describe the bug**

After upgrading from v1.22.x to v1.23.5, all subsequent operations on cluster fail.

`az aks nodepool update --name nodepool1 --cluster-name myaks --resource-group myrg --enable-cluster-autoscaler --min-count 1 --max-count 3`

Results in:

ERROR: (AgentPoolUpgradeVersionNotAllowed) Upgrade agent pool version from 1.23.5 to 1.22.6 is not allowed, you can only upgrade one minor version at a time. Please use [az aks get-upgrades] command to get the supported upgrade paths

`az aks update -g myrg -n myaks --load-balancer-outbound-ips /resource/id`

Results in:

ERROR: (NotAllAgentPoolOrchestratorVersionSpecifiedAndUnchanged) Using managed cluster api, all Agent pools' OrchestratorVersion must be all specified or all unspecified. If all specified, they must be stay unchanged or the same with control plane. 

Indeed, `az aks show -n myaks -g myrg -n myaks` results in:

```
      "currentOrchestratorVersion": "1.23.5",
      "mode": "User",
      "name": "win1",
      "nodeImageVersion": "AKSWindows-2019-containerd-17763.3046.220615",
      "orchestratorVersion": "1.22.6",
```

However, if I use Azure Portal to rescale the nodepool in this state, this fixes the problem.

**To Reproduce**

1. Create cluster w/ k8s v1.22.6 with Windows nodepool.
2. Upgrade k8s version.
3. Try using `ak aks update` to configure outbound IP addresses.

**Expected behavior**

Everything works, like it would earlier. This is a new issue.

**Environment summary**

Running commands in Azure Cloud Shell.

{
  "azure-cli": "2.37.0",
  "azure-cli-core": "2.37.0",
  "azure-cli-telemetry": "1.0.6",
  "extensions": {
    "ai-examples": "0.2.5",
    "aks-preview": "0.5.82",
    "front-door": "1.0.16",
    "resource-graph": "2.1.0",
    "ssh": "1.1.1"
  }
}

**Comment by yonzhan** [comment]
route to CXP team

**Comment by navba-MSFT** [comment]
@wasker Apologies for the late reply. Thanks for reaching out to us and sharing this feedback. We are looking into this issue and we will provide an update.

**Comment by navba-MSFT** [comment]
@wasker I had reached out to the Product Owners in the background and they have confirmed that this is a bug. The fix will be done at the Resource Provider layer and will be released next Monday. Hope this helps.

Tagging the similar issue here https://github.com/Azure/azure-cli/issues/22807

**Comment by navba-MSFT** [comment]
@wasker  I wanted to share a quick update on this. The fix rollout just started yesterday and will take a couple days to reach all regions. Until the fix is deployed to all regions, Maybe you can avoid using version 2022-04-01 for now as a workaround. Hope this helps.

**Comment by navba-MSFT** [comment]
@wasker  I am yet to get confirmation about the fix being deployed to all regions. In the meantime, Could you please try running the same above command again and check if you are facing the issue ?

**Comment by navba-MSFT** [comment]
@wasker The action is currently pending on you to test the command again. If the issue persists, please feel free to reopen this thread. We would be happy to help.
